lib/shared/inference/stability_generator.rb

```diff
@@ -3,12 +3,15 @@
 module ::DiscourseAi
   module Inference
     class StabilityGenerator
-      def self.perform!(prompt, width: nil, height: nil)
+      def self.perform!(prompt, width: nil, height: nil, api_key: nil, engine: nil, api_url: nil)
+        api_key ||= SiteSetting.ai_stability_api_key
```

Comment:
Isn't this a bit of a footgun to keep these in here? Can we just make the three arguments required instead of default `nil`?

Reply:
other apis do support automatic lookup (eg open ai) so I think keeping this aligned is right ... using this from a background thread is a bit of an outlier way of acting.